src/models/tag/stream.rs

```diff
@@ -90,133 +116,157 @@ impl FromIterator<HotTag> for HotTags {
 }
 
 impl HotTags {
-    pub async fn set_global_tag_scores() -> Result<(), DynError> {
-        let mut result;
-        {
-            let graph = get_neo4j_graph()?;
-            let graph = graph.lock().await;
-
-            let query = queries::get::get_global_hot_tags_scores();
-            result = graph.execute(query).await?;
-        }
+    pub async fn reindex() -> Result<(), DynError> {
+        HotTags::get_global_hot_tags(&HotTagsInput {
+            limit: 100,
+            skip: 0,
+            taggers_limit: 20,
+            timeframe: Timeframe::AllTime,
+            tagged_type: Some(TaggedType::Post),
+        })
+        .await?;
+        HotTags::get_global_hot_tags(&HotTagsInput {
+            limit: 100,
+            skip: 0,
+            taggers_limit: 20,
+            timeframe: Timeframe::ThisMonth,
+            tagged_type: Some(TaggedType::Post),
+        })
+        .await?;
+        Ok(())
+    }
 
-        if let Some(row) = result.next().await? {
-            let hot_tags_score: Vec<(f64, &str)> = row.get("hot_tags_score").unwrap_or(Vec::new());
-            let hot_tags_users: Vec<(&str, Vec<String>)> =
-                row.get("hot_tags_users").unwrap_or(Vec::new());
-            // Make sure both list has content before write the indexes
-            if !hot_tags_score.is_empty() && !hot_tags_users.is_empty() {
-                Self::put_index_sorted_set(&TAG_GLOBAL_HOT, hot_tags_score.as_slice(), None, None)
-                    .await?;
-                // Add all the users_id in the SET
-                for (label, user_list) in hot_tags_users.into_iter() {
-                    let values_ref: Vec<&str> = user_list.iter().map(|id| id.as_str()).collect();
-                    Taggers::put_index_set(&[label], &values_ref, None, None).await?;
-                }
-            }
+    fn get_cache_key_parts(tags_query: &HotTagsInput) -> Vec<String> {
+        match &tags_query.tagged_type {
+            Some(tagged) => vec![tags_query.timeframe.to_string(), tagged.to_string()],
+            None => vec![tags_query.timeframe.to_string(), String::from("All")],
         }
-        Ok(())
     }
 
-    pub async fn get_global_tags_stream(
-        skip: Option<usize>,
-        limit: Option<usize>,
-        taggers_limit: Option<usize>,
-    ) -> Result<Option<Self>, DynError> {
-        let hot_tags = Self::try_from_index_sorted_set(
-            &TAG_GLOBAL_HOT,
+    async fn get_from_global_cache(tags_query: &HotTagsInput) -> Result<Option<HotTags>, DynError> {
+        let key_parts = HotTags::get_cache_key_parts(tags_query);
+        let key_parts_vector: Vec<&str> =
+            key_parts.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
+        let data = HotTagsData::try_from_index_json(key_parts_vector.clone().as_slice()).await?;
```

Comment:
But why? We loading the whole json when we had in place everything ALREADY to bring only the wanted values from Redis. Let's stick to the patterns we were using: there was a reason for them.